pkg/aws/v1alpha3/cluster/validate_cluster.go

```diff
@@ -138,6 +143,31 @@ func (v *Validator) ValidateUpdate(request *admissionv1.AdmissionRequest) (bool,
 	return true, nil
 }
 
+func (v *Validator) Cilium(cluster *capi.Cluster) error {
+	release, err := semver.New(key.Release(cluster))
+	if err != nil {
+		return err
+	}
+
+	if !aws.IsCiliumRelease(release) {
```

Comment:
I don't think this is correct.
For new v18 clusters, the annotation will be missing and that is ok.
On the other hand, for v17 clusters being upgraded to v18 we need to enforce it.
Am I missing something?

Reply:
My idea was to check if the release version will be set to v18 or higher. In this function we only check for the updated cluster CR, basically when you edit something, so this means it should already contain the new release version a customer wants to update.

In case we have an update to a lower version we are fine. If the release is v18 or above we try to fetch the annotation from the AWSCluster CR and check if it is set.

Reply:
Maybe I'm missing something.
Let's say I have a new v18 cluster.
The awscluster CR does not have the annotation, and that's ok.
As soon as I try to edit the awscluster, this validation is executed and fails because annotation is missing.
Am I wrong?

Reply:
Ok I think I got it now, so we also need to check what previous release version we had before editing and this must be less thank v18.0.0-alpha

Reply:
exactly I think that's the correct way to do it :+1:

Reply:
So I added some additional tests to see if it works.
When doing a release upgrade we only check the annotation now if we upgrade to > v18 from a pre v18 release.

When upgrading from a pre v18 release to a pre v18 release we don't check the annotation and we also don't check if the current release is already v18 or higher and we upgrade to the next release.

Reply:
Seems ok now